### PR TITLE
Data type list component

### PR DESCRIPTION
**Pull request by mcclowes**
# Description

Introduces a new way of browsing data types at the top level and then again within each data type category.

### CODE REVIEW

**Comment by mcclowes** [comment]
See it in action: https://github.com/codatio/codat-docs/pull/240

**Comment by mcclowes** [comment]
See it in action: https://codat-docs-grf7l9pvd-codat.vercel.app/data-model/all-datatypes

File: docs/data-model/accounting/accounting.md [hunk 1]
```diff
@@ -6,16 +6,24 @@ createdAt: "2020-10-01T13:43:16.106Z"
...
+import DataTypes from '@components/global/DataTypes'
+
 The data model for accounting data sources includes all the accounting and banking data types that you can either:
 
-- Pull, or retrieve, from the Codat API.
-- Push, or submit, to an accounting platform.
+- Get from the Codat API
+- Create, update, and delete from a given accounting platform
+
+## Supported data types
```
Comment (pzaichkina) [inline docs/data-model/accounting/accounting.md:16]: out of interest, do we want the search to happen across data type names only, or also descriptions? e.g. if I search for "transaction", I also get a bunch of hits on the descriptions, which basically means the list isn't really narrowed down at all. 
image

File: docs/data-model/all-datatypes.mdx [hunk 2]
```diff
@@ -0,0 +1,10 @@
+---
+title: "All data types"
+description: "Explore all the types of data you can pull through Codat."
+---
+
+import DataTypes from '@components/global/DataTypes'
+
+Access data from 30+ accounting, ERP, eCommerce, PoS, payments, and banking platforms.
+
+<DataTypes/>
```
Comment (pzaichkina) [inline docs/data-model/all-datatypes.mdx:10]: this is a cool page! is there anything that can be done with styling so that transaction categories look better? also, where do the descriptions come from? the bank feeds one is blank
image

image

File: docs/data-model/accounting/accounting.md (hunk 1 above)
Comment (mcclowes) [reply docs/data-model/accounting/accounting.md:16]: I think in this case it's probably desired? Everything there either is a type of transaction or directly relates to transactions

File: docs/data-model/all-datatypes.mdx (hunk 2 above)
Comment (mcclowes) [reply docs/data-model/all-datatypes.mdx:10]: I can improve on the style in the future, and I'll add a description for the bank feed one

Comment (mcclowes) [reply docs/data-model/all-datatypes.mdx:10]: Amended

File: docs/data-model/accounting/accounting.md (hunk 1 above)
Comment (mcclowes) [reply docs/data-model/accounting/accounting.md:16]: (I can easily exclude the descriptions though, if that makes more sense)

File: sidebars/dataModel.js [hunk 3]
```diff
@@ -1,116 +1,302 @@
...
-  bankFeeds: "/bank-feeds-api#/schemas/",
-};
-
-const composePaths = (schemaPaths, dataModels) => {
-  const categories = Object.keys(schemaPaths);
-
-  const paths = {};
+const dataTypes = [
+  {
+    "key": "chartOfAccounts",
```
Comment (pzaichkina) [inline sidebars/dataModel.js:3]: why did you remove "Aged creditor report", "Aged debtor report", and "Attachment"?

File: src/components/global/DataTypes/dataTypes.js [hunk 4]
```diff
@@ -0,0 +1,319 @@
...
+  },
+]
+
+export const bankfeeds = [
+  {
+    key: "bankfeedsbankaccount",
+    name: "Bank account",
+    schema: "/bank-feeds-api#/schemas/BankFeedBankAccount",
+    product: "bank-feeds",
+    description: "",
```
Comment (pzaichkina) [inline src/components/global/DataTypes/dataTypes.js]:
```suggestion
    description: "The target bank account in a supported accounting package for ingestion into a bank feed.",
```

File: sidebars/dataModel.js (hunk 3 above)
Comment (mcclowes) [reply sidebars/dataModel.js:3]: Ah that's a mistake.

I did realise while doing this though, it's time we stop call our reports data types, they're a different thing

File: docs/data-model/accounting/accounting.md [hunk 5]
```diff
@@ -6,16 +6,26 @@ createdAt: "2020-10-01T13:43:16.106Z"
...
+The following accounting data types are available via Codat's API:
+
+<DataTypes category="accounting"/>
+
+<br/>
+
 ## Integrations setup
 
-For full instructions on how to set up an integration for a specific accounting platform, see the [Accounting integrations](/accounting-api/overview) section.
+For full instructions on how to set up an integration for a specific accounting platform, see the [Accounting integrations](/accounting-api/overview) section.
```
Comment (d-laing) [inline docs/data-model/accounting/accounting.md:31]: I think this should link to /integrations/accounting/overview

File: docs/data-model/accounting/accounting.md [hunk 6]
```diff
@@ -6,16 +6,26 @@ createdAt: "2020-10-01T13:43:16.106Z"
 updatedAt: "2022-11-23T14:23:56.353Z"
 ---
 
+import DataTypes from '@components/global/DataTypes'
+
 The data model for accounting data sources includes all the accounting and banking data types that you can either:
```
Comment (d-laing) [inline docs/data-model/accounting/accounting.md:11]: Suggest removing "either" as its implied by the bulleted list.

The right preposition to use here is really tricky, unless we split into two.

We could use "in" rather than "from":

Create, update, and delete in a given accounting package

Or split into two clauses:

- Create and update in a given accounting package
- Delete from a given accounting package

**Review by d-laing — CHANGES_REQUESTED** [review]
Great stuff! This makes the data model much more findable. 
I made few suggestions to consider.
